cmd/soroban-rpc/internal/methods/get_version_info.go

```diff
@@ -0,0 +1,68 @@
+package methods
+
+import (
+	"context"
+	"github.com/creachadair/jrpc2"
+	"github.com/creachadair/jrpc2/handler"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/config"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/daemon/interfaces"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+)
+
+type GetVersionInfoRequest struct {
+}
+
+type GetVersionInfoResponse struct {
+	Version            string `json:"version"`
+	CommitHash         string `json:"commit_hash"`
+	BuildTimestamp     string `json:"build_time_stamp"`
+	CaptiveCoreVersion string `json:"captive_core_version"`
+	ProtocolVersion    uint32 `json:"protocol_version"`
+}
+
+func NewGetVersionInfoHandler(logger *log.Entry, ledgerEntryReader db.LedgerEntryReader, ledgerReader db.LedgerReader, daemon interfaces.Daemon) jrpc2.Handler {
+	coreClient := daemon.CoreClient()
+	return handler.New(func(ctx context.Context, request GetVersionInfoRequest) (GetVersionInfoResponse, error) {
+
+		var captiveCoreVersion string
+		info, err := coreClient.Info(ctx)
```

Comment:
I would cache this so that we don't invoke a process every time we call the endpoint. To be safe, you can cache it for certain time (e.g. 1 min)

Reply:
@2opremio I called this out [above](https://github.com/stellar/soroban-rpc/pull/132/files#r1560149665) but was even more strict: no RPC method should call out to Captive Core directly since it's not made to handle HTTP requests at scale. We should cache it once on startup/CC restart.

Reply:
Yes I agree with @Shaptic, we have to fetch core version only during bootup and restart. 

I added retry logic with ExponentialBackOff to fetch captive core version during daemon.run()

Reply:
as I mentioned in https://github.com/stellar/soroban-rpc/pull/132#discussion_r1566877185 the core version can change so it's better to cache it for a limited period of time as @2opremio suggested

Reply:
@tamirms oh I totally misunderstood it, I thought RPC process would still restart itself when we change the underlying core. 

I am more inclined towards doing cache part in follow up PR, because `getNetwork` also has the same problem where it talks to core info endpoint. Need to refactor that in same way. if everything else looks okay, I can revert this specific part in order to get the green signal. Thoughts?

Reply:
ok, I think the best solution would be to implement a function in the captive core implementation which exposes the most recent info response from core because we are already periodically querying the info endpoint to expose metrics:

https://github.com/stellar/go/blob/master/ingest/ledgerbackend/captive_core_backend.go#L218-L246

perhaps you could take that approach in a separate PR?

Reply:
Thanks @tamirms  Okay I will target that in separate PR !! For now, I am reverting code to initial approach of calling /info endpoint similar to what we have in `getNetwork`